network/p2p/libp2pNode.go

```diff
@@ -637,3 +642,33 @@ func DefaultPubSub(ctx context.Context, host host.Host, psOption ...pubsub.Optio
 	}
 	return pubSub, nil
 }
+
+// PubsubOption generates a libp2p pubsub.Option from the given context and host
+type PubsubOption func(ctx context.Context, host host.Host) (pubsub.Option, error)
```

Comment:
Option which results in the pubsub.Option, since certain pubsub.Option such as Discovery, needs the libp2p host to be passed in when applying those options.